test/step/nestedStep.js

```diff
@@ -463,4 +463,326 @@ describe('A step', () => {
+                                    description: 'step 3.3',
+                                    steps: [
+                                        { type: "step", description: 'step 3.3.1', steps: null },
+                                        { type: "step", description: 'step 3.3.2', steps: null },
+                                        { type: "step", description: 'step 3.3.3', steps: null }
+                                    ]
+                                }
+                            ]
+                        }
+                    ]
+                })
+            })
+
+        })
+
+        context('returning a value', () => {
+
+            const givenNestedStepsWithNoReturn = () => {
+                const st = step({
+                    'step 1': step(() => { return Ok(1) }),
+                    'step 2': step((ctx) => { 
+                        ctx.stop()
+                        return 2 
+                    }),
+                    'step 3': step(() => { return [3] })
+                })
+                return st
+            }
+
+            it('should run', async () => {
+                //given
+                const st = givenNestedStepsWithNoReturn()
+                //when
+                const ret = await st.run()
+                //then
+                assert.ok(ret.isOk)
+            })
+
+            it('should audit', async () => {
+                //given
+                const st = givenNestedStepsWithNoReturn()
+                //when
+                const ret = await st.run()
+                //then
+                assert.deepStrictEqual(st.auditTrail, {
+                    type: 'step',
+                    description: undefined,
+                    elapsedTime: st.auditTrail.elapsedTime,
+                    steps: [
+                        { type: 'step', description: 'step 1', return: { Ok: 1 }, elapsedTime: st._auditTrail.steps[0].elapsedTime },
+                        { type: 'step', description: 'step 2', return: 2, elapsedTime: st._auditTrail.steps[1].elapsedTime },
+                    ],
+                    return: { Ok: {} }
+                })
+                assert.ok(st.auditTrail.elapsedTime > 0)
+            })
+
+        })
+
+        context('returning Err', () => {
+
+            const givenNestedStepsWithError = () => {
+                const st = step({
+                    'step 1': step(() => { return Ok() }),
+                    'step 2': step((ctx) => {  
+                        ctx.stop()
+                        return Err() 
+                     }),
+                    'step 3': step(() => { return Ok() })
+                })
+                return st
+            }
+
+            it('should run', async () => {
+                //given
+                const st = givenNestedStepsWithError()
+                //when
+                const ret = await st.run()
+                //then
+                assert.ok(ret.isErr)
```

Comment:
It should check the returned value, right?

Reply:
same case